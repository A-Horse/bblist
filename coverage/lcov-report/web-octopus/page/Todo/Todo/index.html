<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for web-octopus/page/Todo/Todo</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../index.html">All files</a> web-octopus/page/Todo/Todo
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">24.49% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>12/49</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">40% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>10/25</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">18.75% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>3/16</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">24.49% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>12/49</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file low" data-value="Todo.js"><a href="Todo.js.html">Todo.js</a></td>
	<td data-value="24.49" class="pic low"><div class="chart"><div class="cover-fill" style="width: 24%;"></div><div class="cover-empty" style="width:76%;"></div></div></td>
	<td data-value="24.49" class="pct low">24.49%</td>
	<td data-value="49" class="abs low">12/49</td>
	<td data-value="40" class="pct low">40%</td>
	<td data-value="25" class="abs low">10/25</td>
	<td data-value="18.75" class="pct low">18.75%</td>
	<td data-value="16" class="abs low">3/16</td>
	<td data-value="24.49" class="pct low">24.49%</td>
	<td data-value="49" class="abs low">12/49</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Sep 10 2017 22:36:46 GMT+0800 (CST)
</div>
</div>
<script src="../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../sorter.js"></script>
</body>
</html>
