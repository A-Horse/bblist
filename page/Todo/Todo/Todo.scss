@import 'style/variable';
@import 'style/mixin';
@import 'style/page/todo/variable';

.todo {
  @include clearfix();
  position: relative;
  padding: $todo-item-padding;
  background-color: $color-background-light;
  border-radius: $radius-middle;
  margin-bottom: 5px;
  transition: height 300ms;
  box-shadow: 0 0 5px #ccc;
  width: 100%;
  opacity: 0.8;
  transition: all 200ms ease-in-out;


  &.done {
    .todo--content {
      text-decoration: line-through;
    }
  }

  .star-checkbox {
    .fa-star-o {

    }
  }



  &.open {
    background-color: $primary-bg;
    opacity: 1;
    color: white;

    .todo-hover-operation {
      visibility: visible;
      float: right;
    }
    .todo--main {
      .star-checkbox {
        opacity: 1;
      }
    }

    .checkbox {
      .fa-square-o {
        color: #fff;
      }
      .fa-check-square-o {
        color: #9ef092;
      }
    }

  }

  &:hover {
    .todo-hover-operation {
      visibility: visible;
    }
  }
}

.todo--main {
  position: relative;
  display: flex;
  justify-content: space-between;

  .checkbox {
    top: 2px;
    flex-shrink: 0;
    height: 25px;
    width: 25px;
    font-size: 20px;
  }

  .star-checkbox {
    margin-right: 5px;
    opacity: 0;


    &.checked {
      opacity: 1;
    }
  }

  .todo-deadline-label {
    color: $white;
    min-width: 60px;
    display: inline-block;
    margin-left: 10px;

    span {
      background-color: $red;
      border-radius: 5px;
      padding-left: 4px;
      padding-right: 4px;
    }
  }
}

.todo--content {
  width: 100%;
  margin-left: 15px;
  margin-right: 10px;
  line-height: 22px;
  user-select: initial;
  vertical-align: middle;
  word-break: break-all;
  font-size: 14px;
}

.todo--content__input {
  width: 100%;
  border: none;
  resize: none;
  border-radius: $radius-middle;
  background-color: $gray-light-depth;
  margin: -3px 13px;
  font-size: 14px;
  height: 22px;
}

.todo-editing--meta {
  margin-top: 10px;

  .todo-editing-field {
    & > .fa {
      margin-right: 10px;
    }
  }


  .todo-editing--deadline {
    display: flex;
    align-items: center;

    .fa-calendar-check-o {
      margin-top: -2px;
    }

    .date-picker {
      margin-left: 10px;
      .date-picker--text {

        &.placeholder {
          color: $white;
        }
      }
    }
  }

  .todo-editing--repeat {
    .select {
      margin-left: 10px;

      svg {
        fill: $white;
        margin-top: -2px;
        margin-left: -5px;
      }

      .select--box {
        border: none;
      }

      .select-box--name {
        color: $white;
      }

      .select--container {
        color: $black;
      }
    }
  }
}

.todo-hover-operation {
  visibility: hidden;
  margin-right: 12px;
  white-space: nowrap;
  width: 50px;
  .icon-delete {
    position: relative;
    top: 3px;
    width: 22px;
  }

  .image-icon {
    margin-left: 10px;
    cursor: pointer;

    &:first-child {
      margin-left: 0;
    }
  }
}
