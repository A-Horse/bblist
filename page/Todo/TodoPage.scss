@import 'style/variable';
@import 'style/mixin';

.todo-page {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  padding-right: 5%;
  padding-bottom: 50px;
  height: 100%;
  box-sizing: border-box;
  padding-top: 42px;
  overflow: hidden;

  .todo-box-container {
    background-color: #f8f8f8;
    padding: 0 10px;;
    display: inline-block;
    height: 100%;
    width: 250px;
    min-width: 250px;
  }

  .todo-list-container {
    min-width: 300px;
    display: inline-block;
    position: relative;
    width: calc(100% - 250px);
    flex: 1;
    height: 100%;
  }

  @media screen and (max-width: 600px) {

    .todo-box-container {
      display: block;
      width: auto;
      min-width: auto;
    }

    .todo-list-container {
      display: block;
      width: auto;
    }

  }
}
