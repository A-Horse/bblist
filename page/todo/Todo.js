import React, {Component} from 'react';
import {connect} from 'react-redux';
import {CheckBox} from 'components/widget/CheckBox';
import DatePicker from 'components/date-picker/DatePicker';
import {getTodoList, updateTodo} from 'actions/todo/todos';
import {MoreIcon} from 'services/svg-icons';
import Textarea from 'react-textarea-autosize';
import {Button} from 'components/widget/Button';

import 'style/page/todo/todo.scss';

class Todo extends Component {
  constructor() {
    super();
  }

  componentWillMount() {
    this.state = {
      editToggle: false
    };
  }

  onDragStart(event) {
    
  }
  
  onDragEnd(event) {
    
  }
  
  finishTodo() {
    
  }
  

  onClick() {
    
  }

  onContentClick() {
    this.setState({editToggle: true});
  }

  updateTodo(event) {
    // TODO extract services  
    const {dispatch, todo} = this.props;
    const newTodo = {content: this.refs.content.value.trim()};
    this.setState({editToggle: false});
    dispatch(updateTodo(todo.id, newTodo)).then(function() {
      // TODO å…ˆ set 
      dispatch(getTodoList());
    });
  }

  onContendChanged(event) {
    if (event.keyCode !== 13) {
      return;
    }
    event.preventDefault();
    this.updateTodo();
  }
    
  render() {
    const {todo} = this.props;
    return (
      <div className='todo' ref='main' onDragStart={this.onDragStart.bind(this)}
           onClick={this.onClick.bind(this)}
           onDragEnd={this.onDragEnd.bind(this)}>

        <div className='todo--main'>
          <CheckBox ref='checkbox'/>
          <p style={{display: !this.state.editToggle ? 'block' : 'none'}} className='todo--content' onClick={this.onContentClick.bind(this)}>{todo.content}</p>
          <Textarea ref='content' onKeyDown={this.onContendChanged.bind(this)} className='todo--content__input' style={{display: this.state.editToggle ? 'block' : 'none'}} defaultValue={todo.content}></Textarea>
          
          <div>
            <MoreIcon className='more-icon'/>
          </div>
        </div>

        <div className='todo-editing--meta' style={{display: this.state.editToggle ? 'block' : 'none'}}>
          <div className='todo-editing--deadline'>
            <label>Deadline:</label>
            <DatePicker ref='date-picker' />
          </div>

          <div>
            <Button styleType='primary' onClick={this.updateTodo.bind(this)}>OK</Button>
          </div>
          
        </div>

      </div>
    );
  }
}

const mapStateToProps = (state) => {
  return {
  };
};

export default connect(mapStateToProps)(Todo);
