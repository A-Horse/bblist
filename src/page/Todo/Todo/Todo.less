
.todo-item {
  width: 100%;
  display: flex;
  flex-wrap: nowrap;

  .todo-done-checkbox {
    width: 30px;
  }

  & > * {
    flex-shrink: 0;
  }

  .todo-item-main {
    width: 100%;
    flex-shrink: inherit;
  }

}
