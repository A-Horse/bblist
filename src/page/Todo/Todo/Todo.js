// @flow
import React, { Component } from 'react';
import { CheckBox } from '../../../components/widget/CheckBox/CheckBox';
import { StarCheckBox } from '../../../components/widget/StarCheckBox/StarCheckBox';
import Textarea from 'react-textarea-autosize';
import { Select, Modal, Icon, Menu, Dropdown, Button, Form, Input, Row, Col, Checkbox } from 'antd';
import { timeout } from '../../../utils/timeout';
import moment from 'moment';
import R from 'ramda';
import { Map } from 'immutable';

import './Todo.scss';

const todoMetaHeight = 24;

import { repeatItems } from '../constants';

class Todo extends Component<
  {
    actions: any,
    todo: Map<Todo>
  },
  {
    editToggle: boolean
  }
> {
  state = {
    editToggle: false
  };

  updateEditingTodo() {
    const newTodo = { content: this.refs.content.value.trim() };
    this.setState({ editToggle: false });
    this.updateTodo(newTodo);
  }

  updateTodo(toPatchData) {
    const { todo } = this.props;
    const data = {
      id: todo.get('id'),
      ...toPatchData
    };
    this.props.actions.UPDATE_TODO_REQUEST(data);
  }

  onContendChanged(event) {
    event.preventDefault();
    this.updateTodo({ content: event.target.value.trim() });
  }

  removeTodo() {
    const { todo } = this.props;
    this.props.actions.DESTORY_TODO_REQUEST({
      id: todo.get('id')
    });
  }

  updateDone = (event: SyntheticEvent<Event>) => {
    this.updateTodo({ isDone: event.target.checked });
  };

  shouldComponentUpdate(nextProps, nextState) {
    if (this.props.todo !== nextProps.todo) {
      return true;
    }
    if (this.state !== nextState) {
      return true;
    }
    return false;
  }

  render() {
    const { todo } = this.props;

    return (
      <div ref="main">
        <div className="todo--main">
          <Checkbox
            className="todo-done-checkbox"
            defaultChecked={todo.get('isDone')}
            onChange={this.updateDone}
          />
          <div
            style={{ display: !this.state.editToggle ? 'block' : 'none' }}
            className="todo--content"
          >
            {todo.get('content')}
            {todo.get('deadline') && (
              <div className="todo-deadline-label">
                <span>{new moment(todo.get('deadline')).format('MM-DD')}</span>
              </div>
            )}
          </div>

          <div className="todo-operation">
            <StarCheckBox
              defaultChecked={todo.get('isStar')}
              onChange={checked => {
                this.updateTodo({ isStar: checked });
              }}
            />
          </div>
        </div>
      </div>
    );
  }
}

export default Todo;
