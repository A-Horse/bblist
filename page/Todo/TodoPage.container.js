import { connect } from 'react-redux';
import TodoPage from 'page/Todo/TodoPage';
import { bindActionCreators } from 'redux';
import { withRouter } from 'react-router-dom';
import Actions, { makeActionRequestCollection } from '../../actions/actions';

const mapStateToProps = state => {
  return {
    todoBoxs: state.todos
      .get('todoBoxIds')
      .map(id => state.todos.get('todoBoxEntities').get(String(id)))
  };
};

const mapDispatchToProps = dispatch => {
  return {
    actions: bindActionCreators(
      makeActionRequestCollection([Actions.GET_TODOBOXS, Actions.ADD_TODOBOX]),
      dispatch
    )
  };
};

const TodoPageContainer = withRouter(connect(mapStateToProps, mapDispatchToProps)(TodoPage));

export default TodoPageContainer;
