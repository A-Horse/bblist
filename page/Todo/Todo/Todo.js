import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { CheckBox } from 'components/widget/CheckBox/CheckBox';
import { StarCheckBox } from 'components/widget/StarCheckBox/StarCheckBox';
import DatePicker from 'components/DatePicker/DatePicker';
import Textarea from 'react-textarea-autosize';
import ClickOutSide from 'components/utils/ClickOutSide';
import { timeout } from 'utils/timeout';
import moment from 'moment';
import R from 'ramda';
import ConfirmModalButton from '../../../components/ConfrimModalButton/ConfirmModalButton';
import { activeTdRepeatHistory, getTodoRepeatHistory } from 'actions/todo/todo-statistics';
import { Select } from 'components/widget/Select';

import './Todo.scss';

const todoMetaHeight = 24;

import { repeatItems } from '../constants';

class Todo extends Component {
  static propTypes = {
    actions: PropTypes.object.isRequired,
    todo: PropTypes.object
  };

  state = {
    editToggle: false
  };

  constructor(props) {
    super(props);
    this.onClickOutside = this.onClickOutside.bind(this);
    this.onRepeatHistoryModal = this.onRepeatHistoryModal.bind(this);
    this.removeTodo = this.removeTodo.bind(this);
    this.updateDone = this.updateDone.bind(this);
    this.updateTodo = this.updateTodo.bind(this);
    this.onContendChanged = this.onContendChanged.bind(this);
  }

  async closeEditable() {
    const currentTodoHeight = this.refs.main.offsetHeight;
    this.refs.main.style.height = currentTodoHeight + 'px';
    await timeout();
    this.refs.main.style.height = currentTodoHeight - todoMetaHeight + 'px';
    this.setState({ editToggle: false });
    await timeout(300);
    this.refs.main.style.height = 'auto';
  }

  async onContentClick() {
    const currentTodoHeight = this.refs.main.offsetHeight;
    this.refs.main.style.height = currentTodoHeight + 'px';
    await timeout();
    this.refs.main.style.height = currentTodoHeight + todoMetaHeight + 'px';
    await timeout(300);
    this.setState({ editToggle: true });
    this.refs.main.style.height = 'auto';
  }

  updateEditingTodo() {
    const newTodo = { content: this.refs.content.value.trim() };
    this.setState({ editToggle: false });
    this.updateTodo(newTodo);
  }

  updateTodo(toPatchData) {
    const { todo } = this.props;
    const data = {
      id: todo.get('id'),
      ...toPatchData
    };
    this.props.actions.UPDATE_TODO_REQUEST(data);
  }

  onContendChanged(event) {
    event.preventDefault();
    this.updateTodo({ content: event.target.value.trim() });
  }

  removeTodo() {
    const { todo } = this.props;
    this.props.actions.DESTORY_TODO_REQUEST({
      id: todo.get('id')
    });
  }

  updateDone(checked) {
    this.updateTodo({ isDone: checked });
  }

  onClickOutside() {
    if (this.state.editToggle) {
      if (this.refs.datePicker.state.toggle) {
        return;
      }
      this.closeEditable();
    }
  }

  onRepeatHistoryModal() {
    this.props.dispatch(activeTdRepeatHistory(this.props.todo.id));
    this.props.dispatch(getTodoRepeatHistory(this.props.todo.id));
  }

  shouldComponentUpdate(nextProps, nextState) {
    if (this.props.todo !== nextProps.todo) {
      return true;
    }
    if (this.state !== nextState) {
      return true;
    }
    return false;
  }

  render() {
    const { todo } = this.props;

    return (
      <ClickOutSide onClickOutside={this.onClickOutside}>
        <div
          className={`todo${this.state.editToggle ? ' open' : ''}${todo.isDone ? ' done' : ''}`}
          ref="main"
        >
          <div className="todo--main">
            <CheckBox
              ref="checkbox"
              defaultChecked={todo.get('isDone')}
              onChange={this.updateDone}
            />
            <div
              style={{ display: !this.state.editToggle ? 'block' : 'none' }}
              className="todo--content"
              onClick={this.onContentClick.bind(this)}
            >
              {todo.get('content')}
              {todo.get('deadline') &&
                !this.state.editToggle && (
                  <div className="todo-deadline-label">
                    <span>{new moment(todo.get('deadline')).format('MM-DD')}</span>
                  </div>
                )}
            </div>
            <Textarea
              ref="content"
              className="todo--content__input"
              style={{ display: this.state.editToggle ? 'block' : 'none' }}
              defaultValue={todo.get('content')}
              onChange={this.onContendChanged}
            />

            <div className="todo-hover-operation">
              <ConfirmModalButton onConfirm={this.removeTodo}>
                <i className="fa fa-trash" aria-hidden="true" />
              </ConfirmModalButton>
              {!!todo.get('repeat') && (
                <i
                  className="fa fa-bar-chart"
                  onClick={this.onRepeatHistoryModal}
                  aria-hidden="true"
                />
              )}
              <StarCheckBox
                defaultChecked={todo.get('isStar')}
                onChange={checked => {
                  this.updateTodo({ isStar: checked });
                }}
              />
            </div>
          </div>
          {this.state.editToggle && (
            <div
              className="todo-editing--meta"
              style={{ display: this.state.editToggle ? 'block' : 'none' }}
            >
              <div className="todo-editing-field todo-editing--deadline">
                <i className="fa fa-calendar-check-o" aria-hidden="true" />
                <label>Deadline:</label>
                <DatePicker
                  ref="datePicker"
                  placeholder="YYYY-MM-DD"
                  hideIcon={true}
                  defaultValue={todo.get('deadline')}
                  onSelected={date => this.updateTodo({ deadline: date ? date.getTime() : null })}
                />
              </div>

              <div className="todo-editing-field todo-editing--repeat">
                <i className="fa fa-repeat" aria-hidden="true" />
                <label>Repeat:</label>
                <Select
                  defaultItem={R.find(R.propEq('value', parseInt(todo.get('repeat'), 10)))(
                    repeatItems
                  )}
                  items={repeatItems}
                  onSelect={repeat => this.updateTodo({ repeat: repeat.value })}
                />
              </div>
            </div>
          )}
        </div>
      </ClickOutSide>
    );
  }
}

export default Todo;
