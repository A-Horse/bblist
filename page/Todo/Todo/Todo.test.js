import React from 'react';
import Todo from './Todo';
import renderer from 'react-test-renderer';
// import ShallowRenderer from 'react-test-renderer/shallow';
import { fromJS } from 'immutable';

test('Todo Component render', () => {
  const todo = fromJS({
    id: 76,
    userId: 2,
    todoBoxId: null,
    content: "I'am a Todo.",
    remark: null,
    isDone: 1,
    isStar: 0,
    repeat: '2',
    deadline: null,
    noticeTime: null,
    doneTime: null,
    tags: null,
    created_at: 1498739573140,
    updated_at: null,
    isDelete: null
  });

  const mockDispatch = jest.fn();
  const tree = renderer.create(<Todo todo={todo} dispatch={mockDispatch} />, {
    createNodeMock: () => document.createElement('textarea')
  });
  expect(tree).toMatchSnapshot();
});
