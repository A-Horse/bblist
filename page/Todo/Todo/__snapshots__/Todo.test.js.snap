// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo Component render 1`] = `
<div>
  <div
    className="todo"
  >
    <div
      className="todo--main"
    >
      <div
        className="checkbox"
        onClick={undefined}
      >
        <label
          className="checkbox--mock"
        >
          <input
            checked={1}
            onChange={[Function]}
            type="checkbox"
          />
          <i
            aria-hidden="true"
            className="fa fa-check-square-o"
          />
        </label>
      </div>
      <div
        className="todo--content"
        onClick={[Function]}
        style={
          Object {
            "display": "block",
          }
        }
      >
        I'am a Todo.
      </div>
      <textarea
        className="todo--content__input"
        defaultValue="I'am a Todo."
        onChange={[Function]}
        onKeyDown={[Function]}
        style={
          Object {
            "display": "none",
            "height": 0,
          }
        }
      />
      <div
        className="todo-hover-operation"
      >
        <img
          className="image-icon"
          onClick={[Function]}
          src="/assets/icons/chart.png"
        />
        <img
          className="image-icon icon-delete"
          onClick={[Function]}
          src="/assets/icons/remove-red.png"
        />
      </div>
      <div
        className="star-checkbox"
        onClick={undefined}
      >
        <label
          className="star-checkbox--mock"
        >
          <input
            checked={0}
            onChange={[Function]}
            type="checkbox"
          />
          <i
            aria-hidden="true"
            className="fa fa-square-o"
          />
        </label>
      </div>
    </div>
  </div>
</div>
`;
